applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_base_element.h

```diff
@@ -0,0 +1,471 @@
+					      std::vector< array_1d<double, 3 > >& rOutput,
+					      const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_ERROR << "CalculateOnIntegrationPoints of the adjoint base condition is called!" << std::endl;
+    }
+
+    void CalculateOnIntegrationPoints(const Variable<array_1d<double, 6 > >& rVariable,
+					      std::vector< array_1d<double, 6 > >& rOutput,
+					      const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_ERROR << "CalculateOnIntegrationPoints of the adjoint base condition is called!" << std::endl;
+    }
+
+    void CalculateOnIntegrationPoints(const Variable<Vector >& rVariable,
+					      std::vector< Vector >& rOutput,
+					      const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_ERROR << "CalculateOnIntegrationPoints of the adjoint base condition is called!" << std::endl;
+    }
+
+    void CalculateOnIntegrationPoints(const Variable<Matrix >& rVariable,
+					      std::vector< Matrix >& rOutput,
+					      const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_ERROR << "CalculateOnIntegrationPoints of the adjoint base condition is called!" << std::endl;
+    }
+
+    void GetValueOnIntegrationPoints(const Variable<double>& rVariable,
+					     std::vector<double>& rValues,
+					     const ProcessInfo& rCurrentProcessInfo) override;
+
+    int Check(const ProcessInfo& rCurrentProcessInfo) override;
+
+    // Sensitivity functions
+
+    void CalculateSensitivityMatrix(const Variable<double>& rDesignVariable, Matrix& rOutput,
+                                            const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateSensitivityMatrix(const Variable<array_1d<double,3>>& rDesignVariable, Matrix& rOutput,
+                                            const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateStressDisplacementDerivative(const Variable<Vector>& rStressVariable,
+                                    Matrix& rOutput, const ProcessInfo& rCurrentProcessInfo);
+
+    void CalculateStressDesignVariableDerivative(const Variable<double>& rDesignVariable, const Variable<Vector>& rStressVariable,
+                                        Matrix& rOutput, const ProcessInfo& rCurrentProcessInfo);
+
+    void CalculateStressDesignVariableDerivative(const Variable<array_1d<double,3>>& rDesignVariable,
+                                            const Variable<Vector>& rStressVariable,
+                                             Matrix& rOutput, const ProcessInfo& rCurrentProcessInfo);
+
+    Element::Pointer pGetPrimalElement()
+    {
+        return mpPrimalElement;
+    }
+    ///@}
+
+    ///@name Public specialized Access - Temporary
+    ///@{
+    ///@}
+
+protected:
+
+    ///@name Protected Lyfe Cycle
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    virtual void AfterPerturbation(const Variable<double>& rDesignVariable,
```

Comment:
Can we eliminate these? This design pattern makes code harder to comprehend even if they are empty 90% of the time. If one really needs to customize a function they can override the function instead of overriding these functions. Besides that function names should start with a verb describing an action.

Reply:
Yes can be eliminated. It is a leftover from when the Shells required some special treatment. Ideally we don't need this at all.

Reply:
Done